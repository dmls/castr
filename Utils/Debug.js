import React from 'react';
import { View, Text } from 'react-native';
import { styles } from '../assets/styles/Styles';

export const print = (data, pretty = true) => {
  return (
    <View style={styles.sectionRow}>
      <Text>{JSON.stringify(data, null, (pretty ? 2 : undefined)) }</Text>
    </View>
  );
};
