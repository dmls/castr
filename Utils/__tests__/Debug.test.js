import React from 'react';
import renderer from 'react-test-renderer';

import { print } from '../Debug';

describe('print function', () => {
  test('renders correctly with default settings', () => {
    const data = { key: 'value' };
    const component = renderer.create(print(data));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with pretty formatting', () => {
    const data = { key: 'value' };
    const component = renderer.create(print(data, true));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly without pretty formatting', () => {
    const data = { key: 'value' };
    const component = renderer.create(print(data, false));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
