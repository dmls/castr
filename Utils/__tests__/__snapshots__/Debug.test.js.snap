// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print function renders correctly with default settings 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 4,
      "marginVertical": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <Text>
    {
  "key": "value"
}
  </Text>
</View>
`;

exports[`print function renders correctly with pretty formatting 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 4,
      "marginVertical": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <Text>
    {
  "key": "value"
}
  </Text>
</View>
`;

exports[`print function renders correctly without pretty formatting 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 4,
      "marginVertical": 10,
      "paddingHorizontal": 10,
    }
  }
>
  <Text>
    {"key":"value"}
  </Text>
</View>
`;
